Guidelines.md

```diff
@@ -110,23 +110,28 @@ This document establishes the guidelines Microsoft REST APIs SHOULD follow so RE
 <!-- /TOC -->
 
 ## 3 Introduction
-Developers access most Microsoft Cloud Platform resources via RESTful HTTP interfaces. Although each service typically provides language-specific frameworks to wrap their APIs, all of their operations eventually boil down to REST operations over HTTP. Microsoft must support a wide range of clients and services and cannot rely on rich frameworks being available for every development environment. Thus a goal of these guidelines is to ensure Microsoft REST APIs can be easily and consistently consumed by any client with basic HTTP support.
+Developers access most Microsoft Cloud Platform resources via HTTP interfaces. Although each service typically provides language-specific frameworks to wrap their APIs, all of their operations eventually boil down to HTTP requests. Microsoft must support a wide range of clients and services and cannot rely on rich frameworks being available for every development environment. Thus a goal of these guidelines is to ensure Microsoft's REST APIs can be easily and consistently consumed by any client with basic HTTP support.
 
-To provide the smoothest possible experience for developers, it's important to have these REST APIs follow consistent design guidelines, thus making using them easy and intuitive. This document establishes the guidelines to be followed by Microsoft REST API developers for developing such RESTful interfaces consistently.
+To provide the smoothest possible experience for developers, it's important to have these APIs follow consistent design guidelines, thus making using them easy and intuitive. This document establishes the guidelines to be followed by Microsoft REST API developers for developing such APIs consistently.
 
 The benefits of consistency accrue in aggregate as well; consistency allows teams to leverage common code, patterns, documentation and design decisions.
 
 These guidelines aim to achieve the following:
-- Define consistent practices and patterns for all REST endpoints across Microsoft.
-- Adhere as closely as possible to accepted REST/HTTP best practices in the industry at-large.
+- Define consistent practices and patterns for all API endpoints across Microsoft.
+- Adhere as closely as possible to accepted REST/HTTP best practices in the industry at-large.*
 - Make accessing Microsoft Services via REST interfaces easy for all application developers.
 - Allow service developers to leverage the prior work of other services to implement, test and document REST endpoints defined consistently.
 - Allow for partners (e.g., non-Microsoft entities) to use these guidelines for their own REST endpoint design.
 
+*Note: The guidelines are designed to align with building services which comply with the REST architectural style, though they do not address or require building services that follow the REST constraints. The term "REST" is used throughout this document to mean services that are in the spirit of REST rather than adhering to REST by the book.*
+
```

Comment:
To me your statement is essentially conveying the same thing. That being
said, I like your version better.
On Mon, Jul 25, 2016 at 12:26 AM Sebastien Lambla <notifications@github.com>
wrote:

> In Guidelines.md
> <https://github.com/Microsoft/api-guidelines/pull/48#discussion_r72018567>
> :
>
> >  - Make accessing Microsoft Services via REST interfaces easy for all application developers.
> >  - Allow service developers to leverage the prior work of other services to implement, test and document REST endpoints defined consistently.
> >  - Allow for partners (e.g., non-Microsoft entities) to use these guidelines for their own REST endpoint design.
> >
> > +*Note: The guidelines are designed to align with building services which comply with the REST architectural style, though they do not address or require building services that follow the REST constraints. The term "REST" is used throughout this document to mean services that are in the spirit of REST rather than adhering to REST by the book.*
> > +
>
> It feels like the note is empty calories, it does not really clarify, id
> think it obfuscates the statement a bit too much. "align with building
> services which comply with ..." means what? It doesnt' preclude nor mandate
> ReST? I think something along the lines of "These guidelines are aimed at
> facilitating the design of HTTP APIs. They do not require implementing the
> various ReST architectural constraints, but aim to be compatible with them.
>
> By the love of all that is architectural in the world, can we *please*
> not have "REST by the book" or "the spirit of REST" or "alignment" and
> other obfuscations in guidelines? They muddy the water even further.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/api-guidelines/pull/48/files/ace225083436c649326df1df09e17f545a90a7f4#r72018567>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAInRJqTNq84dHGG46qKIjDJ_8K5DIxjks5qZGUQgaJpZM4JTOoC>
> .
>